### PR TITLE
FIX: #56 Adds support for multiple Android flavors

### LINKED ISSUE #56: [BUG] nexusIQScan for Android not support multiple flavors
**Describe the bug**
On android we can declare additional "buildTypes" and "productFlavors". Unfortunately scanning does not work when project contain more than one flavor. Gradle raise exception:

> Caused by: org.gradle.api.GradleException: Could not scan the project: Cannot change dependencies of dependency configuration ':app:sonatypeCopyConfiguration' after it has been resolved.

**To Reproduce**
1. Create android project with multiple flavors
2. Connect android library as submodule
3. Run simulated scanning eg. `gradle --stacktrace nexusIQScan`

Sample project containing this error: https://github.com/Shusek/SonarGradlePluginFlavorBug
**Expected behavior**
Plugin should support more than one flavor

**Desktop (please complete the following information):**
 - OS: Linux Ubuntu
 - Gradle Version: 6.7.1
 - JVM Version and Flavor: Android Studio embedded JVM 1.8
 - Nexus IQ Server Version: dry run
 - Version: 2.0.0

**Additional context**
Full stacktrace error:

> `* What went wrong:
> Execution failed for task ':nexusIQScan'.
> > Could not scan the project: Cannot change dependencies of dependency configuration ':app:sonatypeCopyConfiguration' after it has been resolved.
> 
> * Try:
> Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> 
> * Exception is:
> org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':nexusIQScan'.
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:207)
>         at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:263)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:205)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:186)
>         at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:114)
>         at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
>         at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
>         at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
>         at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
>         at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
>         at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
>         at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
>         at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:409)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:399)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:157)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:242)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:150)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:94)
>         at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
>         at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
>         at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
>         at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:356)
>         at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:343)
>         at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:336)
>         at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:322)
>         at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
>         at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
>         at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
>         at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
>         at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
>         at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
>         at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
> Caused by: org.gradle.api.GradleException: Could not scan the project: Cannot change dependencies of dependency configuration ':app:sonatypeCopyConfiguration' after it has been resolved.
>         at org.sonatype.gradle.plugins.scan.nexus.iq.NexusIqScanTask.scan(NexusIqScanTask.java:118)
>         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>         at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
>         at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:49)
>         at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:42)
>         at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:28)
>         at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:726)
>         at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:693)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.run(ExecuteActionsTaskExecuter.java:569)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:395)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:387)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:157)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:242)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:150)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:84)
>         at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:554)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:537)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.access$300(ExecuteActionsTaskExecuter.java:108)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.executeWithPreviousOutputFiles(ExecuteActionsTaskExecuter.java:278)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.execute(ExecuteActionsTaskExecuter.java:267)
>         at org.gradle.internal.execution.steps.ExecuteStep.lambda$execute$1(ExecuteStep.java:33)
>         at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:33)
>         at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:26)
>         at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:67)
>         at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:36)
>         at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:49)
>         at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:34)
>         at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:43)
>         at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:73)
>         at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:54)
>         at org.gradle.internal.execution.steps.CatchExceptionStep.execute(CatchExceptionStep.java:34)
>         at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:44)
>         at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:54)
>         at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:38)
>         at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:49)
>         at org.gradle.internal.execution.steps.CacheStep.executeWithoutCache(CacheStep.java:159)
>         at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:72)
>         at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:43)
>         at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:44)
>         at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:33)
>         at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:38)
>         at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:24)
>         at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:92)
>         at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$0(SkipUpToDateStep.java:85)
>         at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:55)
>         at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:39)
>         at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:76)
>         at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:37)
>         at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:36)
>         at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:26)
>         at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:94)
>         at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:49)
>         at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:79)
>         at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:53)
>         at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:74)
>         at org.gradle.internal.execution.steps.SkipEmptyWorkStep.lambda$execute$2(SkipEmptyWorkStep.java:78)
>         at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:78)
>         at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:34)
>         at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:39)
>         at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:40)
>         at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:28)
>         at org.gradle.internal.execution.impl.DefaultWorkExecutor.execute(DefaultWorkExecutor.java:33)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:194)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:186)
>         at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:114)
>         at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
>         at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
>         at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
>         at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
>         at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
>         at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
>         at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
>         at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:409)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:399)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:157)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:242)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:150)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:94)
>         at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
>         at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
>         at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
>         at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:356)
>         at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:343)
>         at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:336)
>         at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:322)
>         at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
>         at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
>         at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
>         at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
>         at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
>         at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
>         at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
> Caused by: org.gradle.api.InvalidUserDataException: Cannot change dependencies of dependency configuration ':app:sonatypeCopyConfiguration' after it has been resolved.
>         at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.preventIllegalMutation(DefaultConfiguration.java:1128)
>         at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.validateMutation(DefaultConfiguration.java:1099)
>         at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.lambda$validateMutationType$1(DefaultConfiguration.java:298)
>         at org.gradle.internal.ImmutableActionSet$SingletonSet.execute(ImmutableActionSet.java:225)
>         at org.gradle.api.internal.DefaultDomainObjectSet.assertMutableCollectionContents(DefaultDomainObjectSet.java:71)
>         at org.gradle.api.internal.DefaultDomainObjectCollection.add(DefaultDomainObjectCollection.java:252)
>         at org.gradle.api.internal.DelegatingDomainObjectSet.add(DelegatingDomainObjectSet.java:110)
>         at org.gradle.api.internal.artifacts.DefaultDependencySet.add(DefaultDependencySet.java:68)
>         at org.gradle.api.internal.artifacts.DefaultDependencySet.add(DefaultDependencySet.java:36)
>         at org.sonatype.gradle.plugins.scan.common.DependenciesFinder.lambda$getDependencies$10(DependenciesFinder.java:151)
>         at org.gradle.api.internal.CompositeDomainObjectSet.all(CompositeDomainObjectSet.java:158)
>         at org.gradle.api.internal.DelegatingDomainObjectSet.all(DelegatingDomainObjectSet.java:55)
>         at org.sonatype.gradle.plugins.scan.common.DependenciesFinder.getDependencies(DependenciesFinder.java:145)
>         at org.sonatype.gradle.plugins.scan.common.DependenciesFinder.lambda$findResolvedArtifacts$6(DependenciesFinder.java:76)
>         at org.sonatype.gradle.plugins.scan.common.DependenciesFinder.findResolvedArtifacts(DependenciesFinder.java:78)
>         at org.sonatype.gradle.plugins.scan.common.DependenciesFinder.lambda$findModules$8(DependenciesFinder.java:87)
>         at org.gradle.api.internal.DefaultMutationGuard$2.execute(DefaultMutationGuard.java:42)
>         at org.gradle.internal.Actions.with(Actions.java:251)
>         at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator$2$1.run(BuildOperationCrossProjectConfigurator.java:79)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:395)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:387)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:157)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:242)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:150)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:84)
>         at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
>         at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator$2.run(BuildOperationCrossProjectConfigurator.java:76)
>         at org.gradle.internal.Factories$1.create(Factories.java:26)
>         at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.withMutableState(DefaultProjectStateRegistry.java:245)
>         at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.withMutableState(DefaultProjectStateRegistry.java:226)
>         at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator.runProjectConfigureAction(BuildOperationCrossProjectConfigurator.java:73)
>         at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator.access$400(BuildOperationCrossProjectConfigurator.java:32)
>         at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator$1.doRunProjectConfigure(BuildOperationCrossProjectConfigurator.java:67)
>         at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator$BlockConfigureBuildOperation.run(BuildOperationCrossProjectConfigurator.java:121)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:395)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:387)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:157)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:242)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:150)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:84)
>         at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
>         at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator.runBlockConfigureAction(BuildOperationCrossProjectConfigurator.java:64)
>         at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator.allprojects(BuildOperationCrossProjectConfigurator.java:54)
>         at org.gradle.api.internal.project.DefaultProject.allprojects(DefaultProject.java:676)
>         at org.sonatype.gradle.plugins.scan.common.DependenciesFinder.findModules(DependenciesFinder.java:84)
>         at org.sonatype.gradle.plugins.scan.nexus.iq.NexusIqScanTask.scan(NexusIqScanTask.java:78)
> `
>

### PR DESCRIPTION
**Pull request by guillermo-varela**
Adds support for multiple Android flavors.

It relates to the following issue #s:
* Fixes #56 

cc @bhamail / @DarthHater  / @shaikhu

### CODE REVIEW

File: src/integTest/resources/android-multiple-flavors/build.gradle [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  // Mapping between Gradle and Android plugin versions: https://developer.android.com/studio/releases/gradle-plugin
+  def androidGradlePluginVersion = '3.0.0'
+  if (gradle.gradleVersion.compareTo('6.1.1') >= 0) {
+    androidGradlePluginVersion = '4.0.0'
+  } else if (gradle.gradleVersion.compareTo('5.6.4') >= 0) {
+    androidGradlePluginVersion = '3.6.4'
+  } else if (gradle.gradleVersion.compareTo('5.0') >= 0) {
+    androidGradlePluginVersion = '3.3.3'
+  }
+  dependencies {
```
Comment (shaikhu) [inline src/integTest/resources/android-multiple-flavors/build.gradle:17]: minor. could do with a new line before this block.

File: src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java [hunk 2]
```diff
@@ -59,21 +59,21 @@
   private static final String ATTRIBUTES_SUPPORTED_GRADLE_VERSION = "4.0";
 
   public Set<ResolvedDependency> findResolvedDependencies(Project rootProject, boolean allConfigurations) {
-    Configuration copyConfiguration = createCopyConfiguration(rootProject);
-
-    return rootProject.getAllprojects().stream().flatMap(project -> project.getConfigurations().stream())
+    rootProject.getAllprojects().forEach(p -> p.getConfigurations().forEach(c -> {
+      System.out.println(c.getAllArtifacts().size() + ": " + c.getName());
+    }));
```
Comment (shaikhu) [inline src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java:64]: Minor. You can make this an expression lambda by dropping the curly braces "{}"

Comment (shaikhu) [reply src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java:64]: > Minor. You can make this an expression lambda by dropping the curly braces "{}"

Also you can avoid the nested forEach statements by using flatMap e.g.
```
rootProject.getAllprojects().stream()
        .flatMap(p -> p.getConfigurations().stream())
        .forEach(c -> System.out.println(c.getAllArtifacts().size() + ": " + c.getName()));
```

File: src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java [hunk 3]
```diff
@@ -116,7 +116,12 @@ Module buildModule(Project project) {
 
   @VisibleForTesting
   Configuration createCopyConfiguration(Project project) {
-    Configuration copyConfiguration = project.getConfigurations().maybeCreate(COPY_CONFIGURATION_NAME);
+    String configurationName = COPY_CONFIGURATION_NAME;
+    int i = 0;
+    while (project.getConfigurations().findByName(configurationName) != null) {
+      configurationName += ++i;
+    }
```
Comment (shaikhu) [inline src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java:123]: Personal preference but you could switch this with a for loop e.g.
```
for (int i = 0; project.getConfigurations().findByName(configurationName) != null; i++) {
      configurationName += i;
 }
```

File: src/integTest/resources/android-multiple-flavors/build.gradle (hunk 1 above)
Comment (guillermo-varela) [reply src/integTest/resources/android-multiple-flavors/build.gradle:18]: > minor. could do with a new line before dependencies section.

Done: 49f3eb6e60a343ed61f8f69bc2f937f68da4995d

File: src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java (hunk 2 above)
Comment (guillermo-varela) [reply src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java]: > Minor. You can make this an expression lambda by dropping the curly braces "{}"

This was just for local tests during development and should be merged 😅
Removed: 49f3eb6e60a343ed61f8f69bc2f937f68da4995d

File: src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java (hunk 3 above)
Comment (guillermo-varela) [reply src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java]: > Personal preference but I think you can switch this with a for loop e.g.
> 
> ```
> for (int i = 0; project.getConfigurations().findByName(configurationName) != null; i++) {
>       configurationName += i;
>  }
> ```

Done: 49f3eb6e60a343ed61f8f69bc2f937f68da4995d

**Review by shaikhu — APPROVED** [review]
+1

**Pull request merged**